### ISSUE TITLE
Origin Trial Browser Not Supported

### ISSUE THREAD

**Issue by TheMaskMaker** [WICG/floc#98]
I wish to test the floc api in the origin trial, but cannot get even the demo to run. After talking with a few folks on the WICG forum and trying all of the help and guide docs several times, I thought I should post an issue in case others had the same problem and because Mr. Sam Dutton was kind enough to offer help on another issue thread and I did not wish to derail that conversation for my own debugging needs. 

Windows 10
flags enabled
chrome://settings/privacySandbox does not seem to exist
no ad blockers are on 
floc.glitch.me says browser does not support
Chrome Version 89.0.4389.114

**Comment by wanderingrover** [comment]
@michaelkleber - what are some examples of FLoC Ids - are they 4 digit numbers or they contain labels as well? I am getting 4 digit numbers.

**Comment by michaelkleber** [comment]
@wanderingrover Yes, they are numbers with up to 5 digits.  For example, after clearing my history and then visiting https://www.youtube.com/watch?v=dQw4w9WgXcQ, in my Console I see

```
await document.interestCohort();
> {id: "28033", version: "chrome.1.1"}
```

(Note that this is based on starting Chrome with the flags `--enable-features="FederatedLearningOfCohorts:update_interval/10s/minimum_history_domain_size_required/1,FlocIdSortingLshBasedComputation`, which is why I could get a flock after visiting only a single domain.)

**Comment by samdutton** [comment]
@TheMaskMaker Do you get a cohort value using Michael's approach: start Chrome with flags then run `await document.interestCohort();` from the console?

**Comment by TheMaskMaker** [comment]
@samdutton I tried but I do not even get the interestCohort function to exist.

Separately, I have found that chrome://settings/privacySandbox does not exist; it redirects to chrome://settings/ I was wondering if this meant anything to you about what could be going on. I had a colleague try this on a different OS, and he also could not access chrome://settings/privacySandbox. 

I am having him walk through the steps of setting up flags and restarting chrome as well to see if its my computer, OS, or some other variable.

**Comment by TheMaskMaker** [comment]
My colleague was able to get this running on Mac and on vanilla chrome with flags and no sign-in, so I think it might be a windows issue. I'm goin to try another way of executing the flags than the method in the guide.

**Comment by TheMaskMaker** [comment]
@samdutton I got it working! I was able to get it working by running the exe with flags on power shell. For some reason changing the target did not work for me.

**Comment by TheMaskMaker** [comment]
For anyone else on windows with this problem on windows, make sure the flags are all on 1 line, and execute it from a command line interface (I had mixed results with gitbash but powershell worked).

**Comment by samdutton** [comment]
@TheMaskMaker 🎉 Hooray! 

(And thanks for the tips about running on Windows.)

**Comment by SKOCHERI** [comment]
On FloC demo, the cohort ID and version is unavailable for chrome version 91.0.4469.4 (Official Build) dev (x86_64).
Also on the console when checking for await document.interestCohort(), I get  "Uncaught DOMException: Failed to get the interest cohort: either it is unavailable, or the preferences or content settings has denied access."

Have enabled the flags, logged in, no ad blockers. Can you guide me if I am missing something?

On a test page(locahost), I get document.interestCohort() is not a function issue.

**Comment by SKOCHERI** [comment]
Was able to get the demo (https://floc.glitch.me/)  and also get cohort id on a test page.

> On FloC demo(https://floc.glitch.me/) , the cohort ID and version is unavailable for chrome version 91.0.4469.4 (Official Build) dev (x86_64).
> Also on the console when checking for await document.interestCohort(), I get "Uncaught DOMException: Failed to get the interest cohort: either it is unavailable, or the preferences or content settings has denied access."
> 
> Have enabled the flags, logged in, no ad blockers. Can you guide me if I am missing something?
> 
> On a test page(locahost), I get document.interestCohort() is not a function issue.

**Comment by samdutton** [comment]
Hi @SKOCHERI — just to confirm, is [floc.glitch.me](https://floc.glitch.me) working OK for you now?

**Comment by TheMaskMaker** [comment]
Was going to close since my issue is solved, but I see Sam is still responding to someone, so won't close yet. I'll leave it to @samdutton

**Comment by eroncastro** [comment]
I have Chrome 90.0.4430.93 and even with using the flags 

```
--enable-blink-features=InterestCohortAPI 
```
and 

```
--enable-features="FederatedLearningOfCohorts:update_interval/10s/minimum_history_domain_size_required/1,FlocIdSortingLshBasedComputation,InterestCohortFeaturePolicy"
```
does not work.

**Comment by PavelNikitsiuk** [comment]
@eroncastro I've managed to run it on windows with a new chrome user session (Chrome 90.0.4430.93)

`start chrome --user-data-dir="C:/Chrome dev session" --enable-blink-features=InterestCohortAPI --enable-features="FederatedLearningOfCohorts:update_interval/10s/minimum_history_domain_size_required/1,FlocIdSortingLshBasedComputation,InterestCohortFeaturePolicy"`

**Comment by samdutton** [comment]
Hi @eroncastro 

Apologies for delay in responding — try these flags (recently updated: see [floc.glitch.me](https://floc.glitch.me)):

```
--enable-blink-features=InterestCohortAPI 
--enable-features="FederatedLearningOfCohorts:update_interval/10s/minimum_history_domain_size_required/1,FlocIdSortingLshBasedComputation,InterestCohortFeaturePolicy"
```

**Comment by samdutton** [comment]
@neustar-ci 

A few troubleshooting suggestions: 

- What version of Chrome are you using, on what OS? 
- Just to double check, are you following the instructions for running the browser with flags [here](https://www.chromium.org/developers/how-tos/run-chromium-with-flags)?
- What do you see if you call `await document.interestCohort()` from the  console?

Also, make sure:
- You don't have chrome://settings/privacySandbox disabled.
- You're not in Incognito mode.
